pandas/tests/indexing/test_datetime.py

```diff
@@ -8,6 +8,40 @@
 
 class TestDatetimeIndex(object):
 
+    def test_tuple_setitem_with_datetime_tz(self):
```

Comment:
sorry I meant put these cases in a single test (obviously with several sub-sections, and avoid duplicated code within reason)